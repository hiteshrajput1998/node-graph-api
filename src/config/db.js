import mongoose from 'mongoose';

export default async (db) => {
        try {
        const conn = await mongoose.connect('mongodb://127.0.0.1:27017/CollegeDB');

        // const conn = await mongoose.connect(`mongodb://${process.env.MONGOOSE_DB_HOST}:${process.env.MONGOOSE_DB_PORT}/${process.env.MONGOOSE_DB_NAME}`);
        return conn;
    } catch (err) {
        throw new Error(`MongoDB connection error: ${err}`);
    }
};