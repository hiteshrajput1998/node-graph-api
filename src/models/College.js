import mongoose from 'mongoose';

const collegeSchema = new mongoose.Schema({
    collegeName: {
        type: String
    },
    address: {
        type: String
    }
});

const College = new mongoose.model('College', collegeSchema);

export default College;