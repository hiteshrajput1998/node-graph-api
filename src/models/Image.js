import mongoose from 'mongoose';

const imageSchema = new mongoose.Schema({
    path: {
        type: String
    },
    filename: {
        type: String
    },
    mimetype: {
        type: String
    },
    encoding: {
        type: String
    }
});

const Image = new mongoose.model('Image', imageSchema);

export default Image;