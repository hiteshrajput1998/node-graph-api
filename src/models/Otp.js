import mongoose from 'mongoose';

const otpSchema = new mongoose.Schema({
    email: {
        type: String
    },
    otp: {
        type: String
    },
    created: {
        type: String
    },
});

const Otp = new mongoose.model('Otp', otpSchema);

export default Otp;