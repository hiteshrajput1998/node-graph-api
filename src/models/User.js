import mongoose from 'mongoose';

const userSchema = new mongoose.Schema({
    userName: {
        type: String
    },
    password: {
        type: String
    },
    email: {
        type: String
    },
    created: {
        type: String
    }
});

const User = new mongoose.model('User', userSchema);

export default User;