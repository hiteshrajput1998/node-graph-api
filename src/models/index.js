import College from './College';
import User from './User';
import Image from './Image';

export {
    College,
    User,
    Image
};