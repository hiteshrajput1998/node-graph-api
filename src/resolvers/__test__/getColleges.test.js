/* eslint-disable no-undef */
import resolver from '../index';
import models from '../../models';
import db from '../../config/db';

describe('Resolver', () => {

    beforeEach(() => {
        db();
    });

    it('getColleges2', async () => {
        const result = await resolver.collegeResolver.Query.getcolleges(
            null,
            {},
            models
        );
        expect(result).not.toBeUndefined();
        if (result.length > 0) {
            expect(result[0]).toMatchSnapshot({
                collegeName: expect.any(String),
                address: expect.any(String)
            });
        }
    });

    it('getColleges with empty response', async () => {
        const result = await resolver.collegeResolver.Query.getcolleges(
            null,
            {},
            models
        );
        expect(result).not.toBeUndefined();
        expect(result).toEqual([]);
    });
});