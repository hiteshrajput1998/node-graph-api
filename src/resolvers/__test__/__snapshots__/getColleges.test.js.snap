// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resolver getColleges2 1`] = `
Object {
  "address": Any<String>,
  "collegeName": Any<String>,
  "id": "6067055a117b8b2c94ad17c0",
}
`;
